samples/bluetooth/central_iso/src/bt_clock_hal_nrf5.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2024, Florian Grandel, grandcentrix GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ *
+ * @brief Bluetooth clock hardware abstraction for nRF5x
+ *
+ * @details Currently, there is no way to access the Bluetooth clock
```

Comment:
As you have seen earlier, the timestamps in the HCI packets are defined by the controller.
For the Zephyr open source link layer, the timestamp maps directly to RTC0 ticks. For the SoftDevice Controller the timestamps also takes wrapping into account. Therefore, the approach taken here will only work until the RTC wraps (8 minutes). Also, the approach taken here will only work when run on a single-core 52 series device. That is, it will not work on the nRF53 app core.

Looking at this it looks like the only feasible way forward:
* Remove reading out the controller clock
* Never issue SDUs with timestamps.
* Not use timers to trigger submission of SDUs, just push them to controller as quickly as possible.

The drawback will be that the total end-to-end latency becomes slightly longer.
If we want to optimize for reduced latency, controller-specific assumptions are needed.

Reply:
I agree with @rugeGerritsen here

As the current BT Core spec works w.r.t. ISO over HCI, we can't really make any (good) assumptions, and the best we really can do is just to always enqueue at least 2 SDUs and TX as fast as possible. 

There are no guarantees we can make without knowing the specific controller. 

I've recently refactored TX for a similar sample: https://github.com/zephyrproject-rtos/zephyr/pull/73406

Effectively moving a timer-based approach to just sending as fast and much as possible, based on the available buffers. 

Keep in mind that the buffer handling will also change as part of https://github.com/zephyrproject-rtos/zephyr/pull/72090

Reply:
Agreed, see my comment re next steps.